fastlane/lib/fastlane/actions/hipchat.rb

```diff
@@ -46,19 +46,19 @@ def self.run(options)
           end
         else
           ########## running on V2 ##########
+          escaped_channel = URI.unescape(channel) == channel ? URI.escape(channel) : channel
```

Comment:
Do you think we could simply use `channel.shellescape`?

Reply:
Hi @ohayon, many thanks for your review, it is a big pleasure for me to contribute to the project. 

About your comment, `channel.shellescape` encode the string to be a valid shell resource and in this case we need a string encoded to be a valid URL resource, in "code words":

```ruby
require 'shellwords'
require 'uri'

channel = "Room name"
print(channel.shellescape) # prints "Room\\ name"
print(URI.escape(channel)) # prints "Room%20name"
```

The complexity of the line is because encoding an already encoded string produce a new value:

```ruby
print(URI.escape(URI.escape(channel))) # prints "Room%2520name"
```

for that reason we first verify if the `channel` string is already encoded or not.

Reply:
Got it! Thanks for the explanation, that makes good sense. Do you think you could just add a little comment to make it a bit more clear and then we can get this merged in? 🚀  🐙

Reply:
Done!